account-keys/src/entropy_check.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+use displaydoc::Display;
+
+/// Check an entropy string for "obvious" statistical problems or blacklisted values.
+///
+/// Note: It would be good to try to make the checks constant-time, since the root
+/// entropy is generally a secret value.
+pub fn check_root_entropy(root_entropy: &[u8]) -> Result<(), RootEntropyProblem> {
```

Comment:
I'd recommend factoring this into an implementation of the [NIST tests](https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-22r1a.pdf) against an RNG generally, and running it on startup---that is, on startup, grab an RNG and run the tester on it, and then simply refuse to start if it fails. This could happen on node/enclave startup as an environmental sanity check, and would not preclude the use of test vectors.